libraries/AP_PiccoloCAN/AP_PiccoloCAN.cpp

```diff
@@ -362,7 +409,9 @@ void AP_PiccoloCAN::send_esc_messages(void)
     // TODO - How to buffer CAN messages properly?
     // Sending more than 2 messages at each loop instance means that sometimes messages are dropped
 
-    if (hal.util->get_soft_armed()) {
+    // No ESCs are selected? Don't send anything
+    if (_esc_bm == 0x00) {
```

Comment:
It's a bit funny to have an "if" with nothing within the curly braces.  Maybe just make it a !.

Reply:
I've adjusted the logic here to make it more readable